examples/nidaqmx_analog_input/teststand_nidaqmx.py

```diff
@@ -23,6 +23,10 @@ def create_nidaqmx_tasks(sequence_context: Any) -> None:
     with GrpcChannelPool() as grpc_channel_pool:
         teststand_support = TestStandSupport(sequence_context)
         pin_map_id = teststand_support.get_active_pin_map_id()
+
+        if not pin_map_id:
```

Comment:
Is there not a way to change our run time library code such that the logic for the code module doesn't need to change? What errors if you didn't early return here? Or is it because we end up registering a task with no channels, and then that causes weird down stream issues with the measurement?

Reply:
I'm not sure what the implications of doing this in the framework are. This is the error we get:
```
<class 'ValueError'>
No reserved sessions matched instrument type ID 'niDAQmx'. Expected single or multiple sessions, got 0 sessions.

Call Stack information:
  File "C:\Users\vagrant\Desktop\measurementlink-python-examples-1.4.0\nidaqmx_analog_input\teststand_nidaqmx.py", line 37, in create_nidaqmx_tasks
    with reservation.create_nidaqmx_tasks(
  File "C:\Users\vagrant\AppData\Local\Programs\Python\Python39\Lib\contextlib.py", line 119, in __enter__
    return next(self.gen)
  File "C:\Users\vagrant\Desktop\measurementlink-python-examples-1.4.0\nidaqmx_analog_input\.venv\Lib\site-packages\ni_measurementlink_service\session_management\_reservation.py", line 755, in _initialize_sessions_core
    raise ValueError(

ValueError: No reserved sessions matched instrument type ID 'niDAQmx'. Expected single or multiple sessions, got 0 sessions.
```

We're calling into `_reservation.py`'s `create_nidaqmx_tasks`, which calls `_initialize_sessions_core`. That is where the error is raised. We expect our `_reservation` object to have sessions that match niDAQmx. 

Maybe we should make it possible to return an empty array of tasks? I'm not entirely sure.

Reply:
Looking at the LV libraries, it looks like the core reservation code allows for an empty reservation. However, the shared code for creating driver specific sessions from the reservation will throw an error if there are zero matching session infos. For Python, it appears both the reservation class and the driver specific create session APIs will throw an error if zero matching sessions are found.

I'm not entirely sure what the right answer is here, but ideally the solution wouldn't require the user to change how their code is written. Maybe we're OK with this is if it only for DAQmx, but I think it would be better if the user didn't have to concern themselves with this at all. I'm inclined to say we should:
- Loosen up the reservation classes so that zero matching sessions is allowed. Or at least it is allowed for the multiple session reservation class. Throwing an error for the single reservation class still seems reasonable.
- Either do the same for all of the driver specific create session APIs or just make a special concession for the DAQmx specific entry point. I don't know if returning an empty enumerable is problematic with any of these APIs from a Python perspective.
- Live with the overhead of calling `session_management_client.register_sessions` and making a grpc call with an empty set or add some client side logic to early return.